owner-onboarding-server/src/ov_management/mod.rs

```diff
@@ -0,0 +1,363 @@
+    }
+
+    let mut unowned_guid = Vec::new();
+    let mut owned_guid = Vec::new();
+    for (i, ov) in ovs.iter().enumerate() {
+        match validate_ov(ov, &user_data.owner_key) {
+            Ok(state) => {
+                if state {
+                    owned_guid.push(ov.header().guid().to_string());
+                } else {
+                    unowned_guid.push(ov.header().guid().to_string());
+                }
+            }
+            Err(e) => match e {
+                ErrorCode::IncompleteVoucherDescription() => {
+                    let message = ErrorCode::generate(ErrorCode::IncompleteVoucher(
+                        owned_guid.len() + unowned_guid.len(),
+                    ));
+                    return Ok(Box::new(warp::reply::with_status(
+                        warp::reply::json(&message),
+                        StatusCode::BAD_REQUEST,
+                    )));
+                }
+                ErrorCode::InvalidVoucherSignatureDescription() => {
+                    let message = ErrorCode::generate(ErrorCode::InvalidVoucherSignature(i));
+                    return Ok(Box::new(warp::reply::with_status(
+                        warp::reply::json(&message),
+                        StatusCode::BAD_REQUEST,
+                    )));
+                }
+                ErrorCode::PemParseErrorDescription() => {
+                    let message = ErrorCode::generate(ErrorCode::PemParseError(owned_guid.len()));
+                    return Ok(Box::new(warp::reply::with_status(
+                        warp::reply::json(&message),
+                        StatusCode::BAD_REQUEST,
+                    )));
+                }
+                _ => {}
+            },
+        }
+    }
+
+    if !unowned_guid.is_empty() {
+        let message = ErrorCode::generate(ErrorCode::UnwonedVoucher(unowned_guid));
+        return Ok(Box::new(warp::reply::with_status(
+            warp::reply::json(&message),
+            StatusCode::BAD_REQUEST,
+        )));
+    }
+
+    for ov in ovs.iter() {
+        let update_store = user_data
+            .ownership_voucher_store
+            .store_data(ov.clone().header().guid().to_owned(), ov.clone())
+            .await
+            .map_err(|e| e.to_string());
+
+        if let Err(e) = update_store {
+            let message = ErrorCode::generate(ErrorCode::StoreError(e));
+            return Ok(Box::new(warp::reply::with_status(
+                warp::reply::json(&message),
+                StatusCode::INTERNAL_SERVER_ERROR,
+            )));
+        }
+    }
+
+    let uuid = warp::reply::json(&owned_guid);
+    Ok(Box::new(warp::reply::with_status(
+        uuid,
+        StatusCode::CREATED,
+    )))
+}
+
+async fn delete_ov(
+    uuids: Vec<String>,
+    user_data: Arc<OwnerServiceUD>,
+) -> core::result::Result<Box<dyn warp::reply::Reply>, warp::Rejection> {
+    let mut unknown_device = Vec::new();
+    for uuid in &uuids {
+        if Guid::from_str(uuid).is_err() {
```

Comment:
since you'll be needing the unwrapped uuid also below, you can `let uuid = match [...]`  and do the error handling of the uuid first.